### ISSUE TITLE
On damage event triggering with pearls when I checked for players

### ISSUE THREAD

**Issue by BlacKSniper41** [SkriptLang/Skript#4606]
### Skript/Server Version

Skript Version: 2.6
Server Version: 1.17.1

### Bug Description

This is part of the code of my combatlog system:

```
on damage:
    victim is player
    attacker is player
    if {combattime.%victim%} is not 0:
        set {combattime.%victim%} to 25
```

This triggers when I land a pearl becouse I get damaged but it shouldn't trigger becouse I checked for both victim and attacker if it's a player. I might have done something wrong tho, but I have no idea.

### Expected Behavior

The code shouldn't detect pearls as players

### Steps to Reproduce

```
on damage:
    victim is player
    attacker is player
    if {combattime.%victim%} is not 0:
        set {combattime.%victim%} to 25

on every second:
	loop all players:
		if {combattime.%loop-player%} is greater than 0:
			send action bar "&c🗡 You're now in combat! &f(&c%{combattime.%loop-player%}%&f)" to loop-player
			set {%loop-player%.log} to true
			remove 1 from {combattime.%loop-player%}
		else:
			if {combattime.%loop-player%} is set:
				send action bar "&a🗡 You're out of combat! &f(25 seconds)" to loop-player
				set {%loop-player%.log} to false
				delete {combattime.%loop-player%}
```

that's my combatlog code, but the on damage event is the important part.

I tried to write and if statement for checking if damage cause is not fall but that didn't work aswell.

### Errors or Screenshots

I got 0 errors.

### Agreement

- [X] I have read the guidelines above and confirm I am following them with this report.

**Comment by AyhamAl-Ali** [comment]
Tests resulted in this
![image](https://user-images.githubusercontent.com/20037329/154800468-e8e17c9b-83aa-436a-8094-9ad32953975b.png)
```v
on damage:
	broadcast "a: %attacker% %event.getDamager()%"
	broadcast "v: %victim% %event.getEntity()%"
	broadcast "dc: %damage cause%"
```

This is caused by

https://github.com/SkriptLang/Skript/blob/094afa9736489a19b81494b3226f91c0fe74c296/src/main/java/ch/njol/skript/expressions/ExprAttacker.java#L84-L86

This is made to get the attacker from shoot event but can lead into an unwanted result such as this issue's case. I will ask the dev team and make a PR for it

**Comment by TPGamesNL** [comment]
I think if CB decided that this case should be considered fall damage caused by an enderpearl projectile we should keep this behaviour, even though it can be confusing: I don't feel like it'd be a good idea to add a workaround for this specific case. If a dev disagrees please reopen

**Comment by Pikachu920** [comment]
not going to reopen just yet, but just to put it out there I disagree that `attacker` should be the shooter in this case. It makes sense that the damage cause is `fall`, however I don't think that there is an attacker in this case (just like normal fall damage, it's environmental damage even though it is technically self inflicted).

**Comment by TPGamesNL** [comment]
I agree with you on that, but I'm not sure if adding a workaround specifically for fall damage (or in general, environmental damage) with an attacker is a great idea.
